File: tests/fixtures/go/cross_function_taint.go
package main

import "fmt"

func source() string {
	return "tainted"
}

func passthrough(data string) string {
	return data
}

func sink(value string) {
	fmt.Println(value)
}

func main() {
	data := source()
	processed := passthrough(data)
	sink(processed)
}

File: tests/fixtures/rust/cross_function_taint.rs
fn source() -> String {
    String::from("tainted")
}

fn passthrough(data: String) -> String {
    data
}

fn sink(value: &str) {
    println!("{}", value);
}

fn main() {
    let data = source();
    let processed = passthrough(data);
    sink(&processed);
}
